components/arm/server.go

```diff
@@ -130,6 +131,31 @@ func (s *serviceServer) GetGeometries(ctx context.Context, req *commonpb.GetGeom
 	}
 	geometries, err := res.Geometries(ctx, req.Extra.AsMap())
 	if err != nil {
+		// if the error tells us the method is unimplemented, then we
+		// can use the kinematics and joint positions endpoints to
+		// construct the geometries of the arm
+		if errors.Is(err, errors.New("unimplemented")) {
```

Comment:
This still seems fragile.

At the least this should be a constant at the top of the file rather than a string defined in the code body.